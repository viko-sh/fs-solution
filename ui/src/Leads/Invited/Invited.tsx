import React, { Component } from 'react';
import { InvitedLead } from '../Lead';
import axios from 'axios';

type InvitedState = {
  loading: boolean;
  leads: [];
};

export class Invited extends Component<InvitedState, any> {
  state = {
    loading: false,
    leads: []
  };

  async componentDidMount() {
    const res = await axios.get('http://localhost:9000/api/jobs/invited');
    this.setState({
      leads: res.data
    });
  }

  async acceptLead(id: number) {}

  async declineLead(id: number) {}

  render() {
    const { leads } = this.state;
    return (
      <div>
        {leads.length > 0 &&
          leads.map((lead, i) => (
            <InvitedLead {...lead} key={`invited-lead-${i}`} />
          ))}
      </div>
    );
  }
}
