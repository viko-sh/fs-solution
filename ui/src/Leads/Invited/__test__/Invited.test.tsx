import React from 'react';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import { InvitedInner } from '../Invited';

const defaultProps = {
  leads: [
    {
      id: 7,
      contactName: 'Darth',
      price: 30,
      description: 'Praesent elit ',
      categoryName: 'Electrical',
      suburbName: 'Bondi',
      postCode: '2026',
      createdAt: '2019-11-21T18:57:43.000Z'
    },
    {
      id: 4,
      contactName: 'Kylo',
      price: 15,
      description: 'Proin semper ',
      categoryName: 'Handyman',
      suburbName: 'Surry Hills',
      postCode: '2010',
      createdAt: '2019-11-21T18:57:43.000Z'
    }
  ],
  getInvited: jest.fn
};

const mountComponent = (customProps: any = {}) => {
  const props = { ...defaultProps, ...customProps };
  const wrapper = shallow(<InvitedInner {...props} />);
  return { props, wrapper };
};

describe('Invited component', () => {
  it('Should render', done => {
    const { wrapper } = mountComponent();
    expect(wrapper).toHaveLength(1);
    done();
  });
  it('should be match with snapshot', done => {
    const { wrapper } = mountComponent();
    wrapper.instance().componentDidMount();
    expect(wrapper).toMatchSnapshot();
    done();
  });
});
