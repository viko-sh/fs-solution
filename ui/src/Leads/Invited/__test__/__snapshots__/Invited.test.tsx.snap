// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accepted component should be match with snapshot 1`] = `ShallowWrapper {}`;
